dokka-integration-tests/gradle/src/test/kotlin/AndroidComposeIT.kt

```diff
@@ -0,0 +1,158 @@
+import org.jetbrains.dokka.gradle.utils.*
+import org.jetbrains.dokka.gradle.utils.addArguments
+import org.jetbrains.dokka.gradle.utils.build
+import org.jetbrains.dokka.it.gradle.junit.*
+import kotlin.io.path.deleteRecursively
+
+/**
+ * Integration test for the `it-android-compose` project.
+ */
+@TestsDGPv2
+@TestsAndroidGradlePlugin
+@TestsCompose
+@WithGradleProperties(GradlePropertiesProvider.Android::class)
+class AndroidComposeIT {
+
+    @DokkaGradlePluginTest(sourceProjectName = "it-android-compose")
+    fun `generate dokka HTML`(
+        project: DokkaGradleProjectRunner,
+    ) {
+        project.runner
+            .addArguments(
+                "clean",
+                ":dokkaGenerate",
+                "--stacktrace",
+                "--rerun-tasks",
+            )
+            .build {
+                withClue("expect project builds successfully") {
+                    shouldHaveTask(":dokkaGenerate").shouldHaveOutcome(SUCCESS)
+                }
+
+                withClue("expect all dokka workers are successful") {
+                    project
+                        .findFiles { it.name == "dokka-worker.log" }
+                        .shouldForAll { dokkaWorkerLog ->
+                            dokkaWorkerLog.shouldBeAFile()
+                            dokkaWorkerLog.readText().shouldNotContainAnyOf(
+                                "[ERROR]",
+                                "[WARN]",
+                            )
+                        }
+                }
+
+                withClue("expect configurations are not resolved during configuration time") {
+                    output shouldNotContain Regex("""Configuration '.*' was resolved during configuration time""")
+                    output shouldNotContain "https://github.com/gradle/gradle/issues/2298"
+                }
+            }
+
+        withClue("expect the same HTML is generated") {
+            val expectedHtml = project.projectDir.resolve("expectedData/html")
+            val actualHtmlDir = project.projectDir.resolve("build/dokka/html")
+
+            withClue(
+                """
+                expectedHtml: ${expectedHtml.toUri()}
+                actualHtmlDir: ${actualHtmlDir.toUri()}
+                """.trimIndent()
+            ) {
+                val expectedFileTree = expectedHtml.toTreeString()
+                val actualFileTree = actualHtmlDir.toTreeString()
+                withClue((actualFileTree to expectedFileTree).sideBySide()) {
+                    expectedFileTree shouldBe actualFileTree
+
+                    actualHtmlDir shouldBeADirectoryWithSameContentAs expectedHtml
+                }
+            }
+        }
+    }
+
+    @DokkaGradlePluginTest(sourceProjectName = "it-android-compose")
+    fun `Dokka tasks should be build cacheable`(
+        project: DokkaGradleProjectRunner,
+    ) {
+        project.runner
+            .addArguments(
+                ":dokkaGenerate",
+                "--build-cache",
```

Comment:
tests appears to be testing only up-to-date checks on re-run.
why build cache is needed here?

shouldn't we also tests for some "changes" in outputs or inputs?

Reply:
Thanks, good point, I've updated the assertions to test the actual Dokka generation work tasks, not just the `dokkaGenerate` lifecycle task.